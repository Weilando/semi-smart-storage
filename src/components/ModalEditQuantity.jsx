import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Modal.css';

function ModalEditQuantity(props) {
  ModalEditQuantity.propTypes = {
    show: PropTypes.bool.isRequired,
    quantity: PropTypes.number.isRequired,
    buffer: PropTypes.number.isRequired,
    changeAction: PropTypes.func.isRequired,
    closeAction: PropTypes.func.isRequired,
    updateAction: PropTypes.func.isRequired,
  }

  return(
    <div className={'modal '.concat(props.show ? 'show' : 'hide')}>
      <div className="modal-main">
        <span className="close" onClick={props.closeAction}>&times;</span>
        <h1>Change quantity</h1>
        <form onSubmit={props.updateAction} className="Modal">
          <input
            type="number"
            min="0"
            step="0.1"
            className="Modal"
            value={props.buffer}
            onChange={props.changeAction}
          />
          <input
            type="submit"
            className="Modal"
            value="Update quantity"
          />
        </form>
      </div>
    </div>
  );
}

export default ModalEditQuantity;
