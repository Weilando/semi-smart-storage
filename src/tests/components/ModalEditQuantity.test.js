import React from 'react';
import renderer from 'react-test-renderer';
import ModalEditQuantity from '../../components/ModalEditQuantity';

describe('snapshot tests for ModalEditQuantity component', () => {
  it('should render visible ModalEditQuantity component correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const updateAction = jest.fn();

    const tree = renderer.create(
      <ModalEditQuantity
        show={true}
        quantity={4}
        buffer={5}
        closeAction={closeAction}
        changeAction={changeAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render hidden ModalEditQuantity component correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const updateAction = jest.fn();

    const tree = renderer.create(
      <ModalEditQuantity
        show={false}
        quantity={4}
        buffer={5}
        closeAction={closeAction}
        changeAction={changeAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
