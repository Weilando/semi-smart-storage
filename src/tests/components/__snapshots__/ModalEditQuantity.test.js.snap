// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for ModalEditQuantity component should render hidden ModalEditQuantity component correctly 1`] = `
<div
  className="modal hide"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Change quantity
    </h1>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        min="0"
        onChange={[MockFunction]}
        step="0.05"
        type="number"
        value={5}
      />
      <input
        className="Modal"
        type="submit"
        value="Update quantity"
      />
    </form>
  </div>
</div>
`;

exports[`snapshot tests for ModalEditQuantity component should render visible ModalEditQuantity component correctly 1`] = `
<div
  className="modal show"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Change quantity
    </h1>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        min="0"
        onChange={[MockFunction]}
        step="0.05"
        type="number"
        value={5}
      />
      <input
        className="Modal"
        type="submit"
        value="Update quantity"
      />
    </form>
  </div>
</div>
`;
